proposals/12513-pvc-artifact-storage/README.md

```diff
@@ -0,0 +1,1867 @@
+- No S3-specific knowledge required (buckets, endpoints, regions). Only familiar Kubernetes concepts (PVCs, `StorageClasses`)
+
+### Enterprise Considerations
+
+Many enterprises and Kubeflow distributions prefer not to have additional external dependencies. With filesystem storage:
+
+- No separate object storage project to productize and support
+- Artifact handling is part of the KFP codebase (same team, same release cycle)
+- Troubleshooting stays within the KFP domain
+
+### Per-Namespace Isolation and Scaling
+
+In namespace-local mode, each namespace gets its own dedicated artifact server and PVC. This provides:
+
+- **Storage isolation**: Each team's artifacts are physically separated in their own PVC
+- **Independent scaling**: Teams can scale their artifact server horizontally (with RWX storage) and size their PVC based on workload requirements
+- **Per-namespace quotas**: Kubernetes `ResourceQuotas` can enforce storage limits per team
+
+### Path to Local Development
+
+This design enables future work on running full KFP locally without a Kubernetes cluster, as the artifact APIs can point directly to local filesystem storage. This improves developer experience and enables offline development workflows.
+
+### Additional Use Cases
+
+In some environments, organizations may prefer to use existing Kubernetes storage infrastructure without additional services.
+
+By providing PVC-based storage as an alternative, we:
+
+- Offer a KFP-native artifact storage option (no separate storage system)
+- Enable portability between different storage backends for pipelines using KFP's artifact APIs
+
+**When to use filesystem storage:**
+
+- Deployments where eliminating object storage dependency is preferred
+- Environments where Kubeflow distributions or platform providers prefer not to support additional storage systems
+- Multi-tenant deployments requiring per-namespace storage isolation, scaling, and quotas
+- Development and experimentation scenarios
+
+**When NOT to use filesystem storage:**
+
+- High-throughput production workloads with large artifacts
+- Scenarios requiring object storage features (versioning, lifecycle policies, geo-replication)
+- When existing object storage infrastructure is already available and preferred
+
+### Goals
+
+Based on the user story "As a user, I want to provision Kubeflow Pipelines with just a PVC for artifact storage so that I can quickly get started", this KEP aims to:
+
+1. **Add filesystem storage as an additional backend option** alongside S3-compatible and Google Cloud Storage, primarily using PVC but not limited to it
+2. **Enable zero-configuration storage** for experimentation use cases - a KFP server can be installed with just a PVC for artifact storage
+3. **Provide namespace-isolated artifact storage** with proper subject access review guards in multi-user mode
+4. **Allow any Kubernetes access mode to be configured** - KFP passes through the configuration to Kubernetes (RWO default)
```

Comment:
What does this mean?

Reply:
It means KFP won’t interpret or restrict the access mode value. It will forward the exact string the operator configured into the PVC spec.